apps/projects/views.py

```diff
@@ -294,49 +298,87 @@ def raise_exception(self):
 
     def get_context_data(self, **kwargs):
         """Append insights to the template context."""
+
         context = super().get_context_data(**kwargs)
+        idea_blueprints = {"BS", "IC"}
+        map_idea_blueprints = {"MBS", "MIC", "PB"}
 
         modules = self.project.modules.all()
+        active_blueprints = set(modules.values_list("blueprint_type", flat=True))
+
+        enable_ideas = active_blueprints.intersection(idea_blueprints)
+        enable_map_ideas = active_blueprints.intersection(map_idea_blueprints)
+        enable_polls = "PO" in active_blueprints
+        enable_interactive_event = "IE" in active_blueprints
+
         ideas = Idea.objects.filter(module__in=modules)
         map_ideas = MapIdea.objects.filter(module__in=modules)
-        proposals = Proposal.objects.filter(module__in=modules)
         comments = get_all_comments_project(project=self.project)
 
-        ratings_ideas = Rating.objects.filter(idea__in=ideas)
-        ratings_map_ideas = Rating.objects.filter(mapidea__in=map_ideas)
-        ratings_comments = Rating.objects.filter(comment__in=comments)
+        polls = Poll.objects.filter(module__in=modules)
+        votes = Vote.objects.filter(choice__question__poll__in=polls)
+        answers = Answer.objects.filter(question__poll__in=polls)
+        live_questions = LiveQuestion.objects.filter(module__in=modules)
+        likes = Like.objects.filter(livequestion__in=live_questions)
+
+        values = [Rating.POSITIVE, Rating.NEGATIVE]
+        ratings_ideas = Rating.objects.filter(idea__in=ideas, value__in=values)
+        ratings_map_ideas = Rating.objects.filter(
+            mapidea__in=map_ideas, value__in=values
+        )
+        ratings_comments = Rating.objects.filter(comment__in=comments, value__in=values)
+
+        count_comments = comments.count()
+        count_ratings = ratings_comments.count()
+        count_ideas = 0
+        count_poll_answers = 0
+        count_event_questions = 0
 
         creator_objects = [
-            ideas,
-            map_ideas,
-            proposals,
             comments,
-            ratings_ideas,
-            ratings_map_ideas,
             ratings_comments,
         ]
+
+        if enable_ideas:
+            creator_objects.extend([ideas, ratings_ideas])
+            count_ideas += ideas.count()
+            count_ratings += ratings_ideas.count()
+
+        if enable_map_ideas:
+            creator_objects.extend([map_ideas, ratings_map_ideas])
+            count_ideas += map_ideas.count()
+            count_ratings += ratings_map_ideas.count()
+
+        if enable_polls:
+            count_poll_answers += votes.count() + answers.count()
```

Comment:
I think here we also need to add votes and answers to creator_objects.

Reply:
True, will add them.